src/plonk/lookup/prover.rs

```diff
@@ -0,0 +1,712 @@
+use super::super::{
+    circuit::{Advice, Any, Aux, Column, Fixed},
+    ChallengeBeta, ChallengeGamma, ChallengeTheta, ChallengeX, Error, ProvingKey,
+};
+use super::{Argument, Proof};
+use crate::{
+    arithmetic::{eval_polynomial, parallelize, BatchInvert, Curve, CurveAffine, FieldExt},
+    poly::{
+        commitment::{Blind, Params},
+        multiopen::ProverQuery,
+        Coeff, EvaluationDomain, ExtendedLagrangeCoeff, LagrangeCoeff, Polynomial, Rotation,
+    },
+    transcript::{Hasher, Transcript},
+};
+use ff::Field;
+use std::{collections::BTreeMap, convert::TryFrom, iter};
+
+#[derive(Clone, Debug)]
+pub(crate) struct Permuted<C: CurveAffine> {
```

Comment:
I don't think any of these types should be `Clone` and in the permutation argument I believe it was a mistake that the `Committed` structure implemented `Clone`. (Opened #83 to deal with that and other random things I see.)